File: SDKs/amazonaws.com/groundstation/2019-05-23/java/lib/src/main/java/openapisdk/models/shared/AntennaDownlinkDemodDecodeConfig.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * AntennaDownlinkDemodDecodeConfig
 * Information about how AWS Ground Station should conﬁgure an antenna for downlink demod decode during a contact.
**/
public class AntennaDownlinkDemodDecodeConfig {
    @JsonProperty("decodeConfig")
    public DecodeConfig decodeConfig;
    public AntennaDownlinkDemodDecodeConfig withDecodeConfig(DecodeConfig decodeConfig) {
        this.decodeConfig = decodeConfig;
        return this;
    }
    @JsonProperty("demodulationConfig")
    public DemodulationConfig demodulationConfig;
    public AntennaDownlinkDemodDecodeConfig withDemodulationConfig(DemodulationConfig demodulationConfig) {
        this.demodulationConfig = demodulationConfig;
        return this;
    }
    @JsonProperty("spectrumConfig")
    public SpectrumConfig spectrumConfig;
    public AntennaDownlinkDemodDecodeConfig withSpectrumConfig(SpectrumConfig spectrumConfig) {
        this.spectrumConfig = spectrumConfig;
        return this;
    }
}

File: SDKs/amazonaws.com/groundstation/2019-05-23/python/src/sdk/models/shared/antennadownlinkdemoddecodeconfig.py
import dataclasses
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import decodeconfig as shared_decodeconfig
from ..shared import demodulationconfig as shared_demodulationconfig
from ..shared import spectrumconfig as shared_spectrumconfig


@dataclass_json
@dataclasses.dataclass
class AntennaDownlinkDemodDecodeConfig:
    r"""AntennaDownlinkDemodDecodeConfig
    Information about how AWS Ground Station should conﬁgure an antenna for downlink demod decode during a contact.
    """
    
    decode_config: shared_decodeconfig.DecodeConfig = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('decodeConfig') }})
    demodulation_config: shared_demodulationconfig.DemodulationConfig = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('demodulationConfig') }})
    spectrum_config: shared_spectrumconfig.SpectrumConfig = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('spectrumConfig') }})
    
